src/validations/rules/Writing_FedRAMP_Validations.md

```diff
@@ -0,0 +1,375 @@
+# How to create FedRAMP OSCAL validation constraints
+
+FedRAMP OSCAL documents are structured XML documents. The document syntax is defined by OSCAL XML Schema. However, XML Schema cannot express many semantic constraints, and an additional schema language is employed: Schematron.
+
+# Technologies used
+
+- [XML](https://www.w3.org/standards/xml/) is a structured document langauge.
+- [Namespaces in XML](https://www.w3.org/TR/xml-names/) allow element (and attribute) names to be declared in distinct namespaces
+- [OSCAL](https://pages.nist.gov/OSCAL/) is used to define the several FedRAMP document types in a structured data form. NIST provides XML Schema documents for the various OSCAL document types.
+- [Schematron](https://schematron.com/) is used to declare assertions for documents to be validated
+- [XPath](https://www.w3.org/TR/xpath-31/) is used in Schematron to define rule contexts and assertion tests
+- [XSpec](https://github.com/xspec/xspec) is used to create unit tests for Schematron assertions
+- [XSLT](https://www.w3.org/TR/xslt-30/) is used to manipulate XML documents, and is used to transpile Schematron documents into XSLT documents
+
+Use of Schematron will require familiarity with XPath for context identification and assertion tests, and XSpec for creation of unit tests.
+
+## Useful resources
+
+[xpather.com](http://xpather.com/) is a browser-based XPath exploration tool. It is not a strict XPath tool: namespaces in the XML document are ignored, and the XPath queries must not use include namespace prefixes. This design makes rapid prototyping easier, but readily copy-pasting XPath into Schematron not error-free.
+
+[XSLT Fiddle](https://xsltfiddle.liberty-development.net/) is a browser-based XSLT exploration tool.
+
+# FedRAMP OSCAL References
+
+NIST has designed OSCAL to be flexible to the needs of different authorization and assessment frameworks, whether or not they originate from NIST's very own [Risk Management Framework](https://csrc.nist.gov/projects/risk-management/about-rmf) or not. OSCAL has a set a syntax. For required or recommended data that must be structured and organized in a specific way within this syntax, OSCAL developers use constraints. In OSCAL, constraints and the constraints model define, given syntactically correct documents, what data is required or recommended.
+
+FedRAMP has adopted and extended NIST OSCAL document types. These adaptations are documented in the GSA [fedramp_automation repository](https://github.com/GSA/fedramp-automation/tree/master/documents).
```

Comment:
Why did FedRAMP extend the NIST OSCAL document types?   Maybe "FedRAMP has adopted and extended NIST OSCAL document types in order to support FedRAMP specific requirements".

Reply:
Intent added.